core/dictgen/src/rootcling_impl.cxx

```diff
@@ -3738,6 +3744,10 @@ gOptMultiDict("multiDict",
              llvm::cl::desc("If this library has multiple separate LinkDef files."),
              llvm::cl::cat(gRootclingOptions));
 static llvm::cl::opt<bool>
+gOptNoUsingGlobalStd("noUsingStd",
```

Comment:
@pcanal, is that a good name for a flag? Alternatives could be `noUsingNamespaceStd`, `skipEmitUsingStd`...

Reply:
another alternative ```withoutUsingNamespaceStdStatement``` ... (wink)

The challenge with all those options is that to read them correctly one has to 'remember' that it is not using the work "using" in the colloquial sense but as a C++ technical term ... 

Another alternative ```skipGlobalUsingStdStatement```

@Axel-Naumann Any better ideas?

Reply:
@pcanal, `noGlobalUsingStd` or `declareGlobalUsingStd`?

Reply:
> `noGlobalUsingStd`

Seems like a decent compromise between brevity and clarity.

On a different note, if there is "upstream" any module that was build `withGlobalUsingStd` (i.e. current default) and that module is loaded by rootcling, does that "using namespace std" leak into the current state (i.e. ignore the `noGlobalUsingStd` because the module used did not have it)?

Reply:
Ok, so noGlobalUsingStd is what we want.

I do not know the answer if another module is built with the using clause on. My guess is that since modules are built in isolation it will not affect the module in flight.

Reply:
I guess before we can commit to a new we ought to check that "since modules are built in isolation it will not affect the module in flight."

Reply:
@pcanal Not sure what do you mean with the modules built upstream: the ``using namespace std;`` is added in the dictionary cxx file global scope (I guess this already means that the modules are independent) and it will be added in any case, with or w/o this ``noGlobalUsingStd`` option. The problem pops up only because currently it is added before the includes added by GenerateNecessaryIncludes.

So, should I change the option name to ``noGlobalUsingStd``?

Reply:
Ah, that’s a fair point, @pcanal, we change the dictionary file content (G__) which has nothing to do with the using namespace std inside rootcling.

Reply:
I see .. The scope of the option is much more limited than I thought.

So stepping back for a second.  What is the (higher level) purpose of PR?  How does it help?

Thanks.

Reply:
@pcanal : see this: https://sft.its.cern.ch/jira/browse/ROOT-10661

Reply:
Thanks for your clarification.

Indeed I would vote for "noGlobalUsingStd".  Thanks.

Reply:
Ok, changed to ``noGlobalUsingStd``.
Cheers,
 Ruben